src/test/java/org/jenkinsci/plugins/workflow/cps/CpsThreadTest.java

```diff
@@ -67,7 +67,8 @@
         List<CauseOfInterruption> causes = iba.getCauses();
         assertEquals(1, causes.size());
         assertEquals(CauseOfInterruption.UserInterruption.class, causes.get(0).getClass());
-        r.assertLogContains("never going to stop", b); // TODO under heavy load sometimes fails here; why?
+        r.waitForMessage("Finished: ABORTED", b); // TODO WorkflowRun.isBuilding() can go to false before .finish has completed
```

Comment:
@jglick - Either way, you'll file a bug on, right?  Thanks.
That was my only point - that the TODO is noting behavior that is a bug.